morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -702,44 +703,49 @@ public void setQueryFactory(final QueryFactory queryFactory) {
 
     @Override
     public <T> Iterable<Key<T>> save(final Iterable<T> entities) {
-        if (entities == null) {
-            return new ArrayList<Key<T>>();
-        }
         Iterator<T> iterator = entities.iterator();
-        if (!iterator.hasNext()) {
-            return new ArrayList<Key<T>>();
-        }
-        return save(entities, getWriteConcern(iterator.next()));
+        return !iterator.hasNext()
+               ? Collections.<Key<T>>emptyList()
+               : save(entities, getWriteConcern(iterator.next()));
     }
 
     @Override
     public <T> Iterable<Key<T>> save(final Iterable<T> entities, final WriteConcern wc) {
+        return save(entities, new InsertOptions().writeConcern(wc));
+    }
+
+    @Override
+    public <T> Iterable<Key<T>> save(final Iterable<T> entities, final InsertOptions options) {
         final List<Key<T>> savedKeys = new ArrayList<Key<T>>();
         for (final T ent : entities) {
```

Comment:
I was actually  thinking about that in terms of resolving #1052.  This could be converted to just *always* use the bulk write operation and remove that flag altogether.  Is that what you're thinking, too?